internal/auth/provider/github/github.go

```diff
@@ -0,0 +1,80 @@
+package github
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/google/go-github/v51/github"
+	"github.com/google/uuid"
+	"github.com/rapidmidiex/rmx/internal/auth"
+	"golang.org/x/oauth2"
+
+	ghOAuth "golang.org/x/oauth2/github"
+
+	"github.com/zitadel/oidc/v2/pkg/client/rp"
+	httphelper "github.com/zitadel/oidc/v2/pkg/http"
+	"github.com/zitadel/oidc/v2/pkg/oidc"
+)
+
+var (
+	key = []byte("test1234test1234")
+
+	authURI     = "/github"
+	callbackURI = "/github/callback"
+)
+
+func NewGithub(cfg *auth.ProviderCfg) (*auth.Provider, error) {
+	ah, ch, err := initProvider(cfg.ClientID, cfg.ClientSecret)
+	if err != nil {
+		return nil, err
+	}
+
+	return &auth.Provider{
+		AuthHandler:     ah,
+		CallbackHandler: ch,
+		AuthURI:         authURI,
+		CallbackURI:     callbackURI,
+	}, nil
+}
+
+func initProvider(clientID, clientSecret string) (ah http.HandlerFunc, ch http.HandlerFunc, err error) {
+	rpConfig := &oauth2.Config{
+		ClientID:     clientID,
+		ClientSecret: clientSecret,
+		RedirectURL:  fmt.Sprintf("http://localhost:8000/v0/auth%v", callbackURI),
```

Comment:
We'll need to pass in the API base as well.
```suggestion
		RedirectURL:  fmt.Sprintf("%s/auth%s", apiBase, callbackURI),
```